### PR TITLE
Adjustments based on discussion in #183

### PR DESCRIPTION
**Pull request by ntninja**
General notes:
  * Generated binaries by CMake are now named `nxengine-evo` and `nxextract` to make them less generic and mimic that the `build-scripts/unix.install-common.sh` script was already doing
  * CMake now also will install the XDG metadata on non-Apple, non-Android Unix targets (i.e. Linux, BSD)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -101,6 +103,7 @@ ENDIF()
 add_definitions("-Wall")
 
 add_executable(nx ${SOURCES})
+set_property(TARGET nx PROPERTY OUTPUT_NAME nxengine-evo)
```
Comment (isage) [inline CMakeLists.txt:106]: This breaks vita and osx packaging.

File: CMakeLists.txt [hunk 2]
```diff
@@ -175,11 +178,17 @@ ENDIF()
 IF(PLATFORM STREQUAL "pc")
     add_executable(extract ${EXTR_SOURCES})
     target_link_libraries(extract ${SDL2_LIBRARY})
+    set_property(TARGET extract PROPERTY OUTPUT_NAME nxextract)
```
Comment (isage) [inline CMakeLists.txt:181]: This breaks osx packaging.

**Review by isage — COMMENTED** [review]
Otherwise looks good.

File: CMakeLists.txt (hunk 2 above)
Comment (ntninja) [reply CMakeLists.txt:181]: Can I just update it? Claiming a generic name such as `extract` in `/usr/{local/,}bin` is really not great upstream behaviour on our side.

**Comment by isage** [comment]
Yeah, sure

File: CMakeLists.txt (hunk 1 above)
Comment (ntninja) [reply CMakeLists.txt:106]: How does it break their packaging? Don't they link this one in statically when building the final image anyways?

**Comment by ntninja** [comment]
This should hopefully fix any issues (albeit untested) you mentioned. Also in the future you'll need to bump the version in `platform/xdg/org.nxengine.nxengine_evo.appdata.xml` as well or people on the Linux builds won't see the new version in some places (see last commit, it's not hard).

**Comment by isage** [comment]
Yeah, i haven't changed it, because 1) there's no release date yet 2) i
wasn't sure if i need to change existing line or add a new one

**Pull request merged**